src/types/ut_types/UtilsTests.cpp

```diff
@@ -89,3 +96,157 @@ void UtilsTests::TestGuidToString()
+    _VerifyXTermColorResult(L"rgb:12/345/678", RGB(0x12, 0x34, 0x67));
+    _VerifyXTermColorResult(L"rgb:123/456/789", RGB(0x12, 0x45, 0x78));
+    _VerifyXTermColorResult(L"rgb:123/4564/789", RGB(0x12, 0x45, 0x78));
+    _VerifyXTermColorResult(L"rgb:123/4564/7897", RGB(0x12, 0x45, 0x78));
+    _VerifyXTermColorResult(L"rgb:1231/4564/7897", RGB(0x12, 0x45, 0x78));
+
+    _VerifyXTermColorResult(L"#111", RGB(0x10, 0x10, 0x10));
+    _VerifyXTermColorResult(L"#123456", RGB(0x12, 0x34, 0x56));
+    _VerifyXTermColorResult(L"#123456789", RGB(0x12, 0x45, 0x78));
+    _VerifyXTermColorResult(L"#123145647897", RGB(0x12, 0x45, 0x78));
+
+    _VerifyXTermColorResult(L"orange", RGB(255, 165, 0));
+    _VerifyXTermColorResult(L"dark green", RGB(0, 100, 0));
+    _VerifyXTermColorResult(L"medium sea green", RGB(60, 179, 113));
+    _VerifyXTermColorResult(L"LightYellow", RGB(255, 255, 224));
+
+    // Invalid sequences.
+    _VerifyXTermColorInvalid(L"");
+    _VerifyXTermColorInvalid(L"r:");
+    _VerifyXTermColorInvalid(L"rg:");
+    _VerifyXTermColorInvalid(L"rgb:");
+    _VerifyXTermColorInvalid(L"rgb:/");
+    _VerifyXTermColorInvalid(L"rgb://");
+    _VerifyXTermColorInvalid(L"rgb:///");
+    _VerifyXTermColorInvalid(L"rgb:1");
+    _VerifyXTermColorInvalid(L"rgb:1/");
+    _VerifyXTermColorInvalid(L"rgb:/1");
+    _VerifyXTermColorInvalid(L"rgb:1/1");
+    _VerifyXTermColorInvalid(L"rgb:1/1/");
+    _VerifyXTermColorInvalid(L"rgb:1/11/");
+    _VerifyXTermColorInvalid(L"rgb:/1/1");
+    _VerifyXTermColorInvalid(L"rgb:1/1/1/");
+    _VerifyXTermColorInvalid(L"rgb:1/1/1/1");
+    _VerifyXTermColorInvalid(L"rgb:this/is/invalid");
+    _VerifyXTermColorInvalid(L"rgba:1/1/1");
+    _VerifyXTermColorInvalid(L"rgbi:1/1/1");
+    _VerifyXTermColorInvalid(L"cmyk:1/1/1/1");
+    _VerifyXTermColorInvalid(L"rgb#111");
+    _VerifyXTermColorInvalid(L"rgb:#111");
+    _VerifyXTermColorInvalid(L"rgb:rgb:1/1/1");
+    _VerifyXTermColorInvalid(L"rgb:rgb:#111");
+    _VerifyXTermColorInvalid(L"#");
+    _VerifyXTermColorInvalid(L"#1");
+    _VerifyXTermColorInvalid(L"#1111");
+    _VerifyXTermColorInvalid(L"#11111");
+    _VerifyXTermColorInvalid(L"#1/1/1");
+    _VerifyXTermColorInvalid(L"#11/1/");
+    _VerifyXTermColorInvalid(L"#1111111");
+    _VerifyXTermColorInvalid(L"#/1/1/1");
+    _VerifyXTermColorInvalid(L"#rgb:1/1/1");
+    _VerifyXTermColorInvalid(L"#111invalid");
+    _VerifyXTermColorInvalid(L"#invalid111");
+    _VerifyXTermColorInvalid(L"#1111111111111111");
+    _VerifyXTermColorInvalid(L"12/34/56");
+    _VerifyXTermColorInvalid(L"123456");
+    _VerifyXTermColorInvalid(L"rgb：1/1/1");
+    _VerifyXTermColorInvalid(L"中文rgb:1/1/1");
+    _VerifyXTermColorInvalid(L"rgb中文:1/1/1");
+    _VerifyXTermColorInvalid(L"这是一句中文");
+    _VerifyXTermColorInvalid(L"RGBİ1/1/1");
+    _VerifyXTermColorInvalid(L"rgbİ1/1/1");
+    _VerifyXTermColorInvalid(L"rgbİ:1/1/1");
+    _VerifyXTermColorInvalid(L"rgß:1/1/1");
+    _VerifyXTermColorInvalid(L"rgẞ:1/1/1");

```

Comment:
how about `rgb:111111111`? it passes the length check but doesn't have any delimiters..